tests/tests_twobody/test_events.py

```diff
@@ -207,3 +207,101 @@ def test_penumbra_event_crossing():
     )
 
     assert expected_penumbra_t.isclose(epoch + penumbra_event.last_t, atol=1 * u.s)
+
+
+def test_orbit_propagation_continues_if_events_terminal_is_False():
+    r = [-6142438.668, 3492467.56, -25767.257] << u.km
+    v = [505.848, 942.781, 7435.922] << u.km / u.s
+    orbit = Orbit.from_vectors(Earth, r, v)
+
+    thresh_lat = 60 * u.deg
+    # Event occurs at ~1701.7 s.
+    latitude_cross_event = LatitudeCrossEvent(orbit, thresh_lat, terminal=False)
+    events = [latitude_cross_event]
+
+    # The last two tofs are after the detection of the event.
+    tofs = [1000, 1250, 1500, 1710, 2000] << u.s
+    rr, _ = cowell(
+        Earth.k,
+        orbit.r,
+        orbit.v,
+        tofs,
+        events=events,
+    )
+
+    # Check position vector of orbit doesn't hang.
+    assert not np.allclose(rr[-1], rr[-2])
```

Comment:
This assertion is a bit confusing. What about checking that the. `len` of `rr` is the same as the original `tofs`? Also, I'm wondering if the norm of `rr` becomes zero after "lithobrake" and if it's something we can use here as well.

Reply:
I wanted to confirm, through this test, the position vectors don't hang in the case where the event was detected and the terminal was set to False. Earlier, `len(rr) == len(tofs)` was satisfied, but the position vectors hanged (i.e. kept repeating till `tofs[-1]`). Would you recommend making the current assertion condition a bit easier to read?

Reply:
> I'm wondering if the norm of `rr` becomes zero after "lithobrake" and if it's something we can use here as well.

Does it mean adding some more checks that check something with `rr` for all the added event tests?

Reply:
Checking the length of vectors, as claimed by @Yash-10, does not ensure no repeated position vectors. The last values of the array need to be checked directly.

Reply:
I understand now, thanks for the clarification!